### PR TITLE
COSA-1094: Add drop-down for utm_source and utm_medium

### PR DESCRIPTION
**Pull request by JuanVqz**
[COSA-1094](https://ombulabs.atlassian.net/browse/COSA-1094)

[COSA-1094]: https://ombulabs.atlassian.net/browse/COSA-1094?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by JuanVqz** [comment]
@FionaDL @etagwerker 
After creating a `Link` record it automatically creates `social_media_snippets` but we ran into the following error:


```
# => {"error"=>{"message"=>"You exceeded your current quota, please check your plan and billing details.", "type"=>"insufficient_quota", "param"=>nil, "code"=>nil}}
```

We need to verify we don't run out of quotas before getting a successful response, wondering if we should create an issue or how to proceed.
Also, may it be the token I generate?

**Comment by torresga** [comment]
@JuanVqz overall I think this looks good and I would like to approve this PR. However, I did notice an issue and I was wondering how I should proceed. I noticed when I create the shares, an extra `https://` is added in front of the resulting link. Do you think I can approve this PR and then open up a new issue for that? 

On second glance, I am unsure whether those links should be shortened. I don't have an API key for Rebrandly. Maybe that's why I am getting this issue?

Screen Shot 2023-04-25 at 3 44 42 PM

**Comment by JuanVqz** [comment]
> 

Good questions, I'm not pretty sure but I can check it tomorrow and let you know :)

**Comment by etagwerker** [comment]
@JuanVqz It's probably related to your token. The token we are using in production has not hit the monthly limit: 

Screen Shot 2023-04-26 at 10 25 02 AM

While in development and test mode, we should not be hitting the API, it should only happen in production. If you see it happening in development or test mode, please open an issue or JIRA task so we can fix that problem.

File: app/models/share.rb [hunk 1]
```diff
@@ -1,6 +1,9 @@
 class Share < ApplicationRecord
   belongs_to :link
 
+  UTM_SOURCES = %w[a b]
+  UTM_MEDIUMS = %w[c d]
```
Comment (etagwerker) [inline app/models/share.rb]: @JuanVqz If you don't know what to put here, you can check out our UTM spreadsheet.

**Review by etagwerker — CHANGES_REQUESTED** [review]

File: app/models/share.rb (hunk 1 above)
Comment (JuanVqz) [reply app/models/share.rb]: Hey good suggestion, we grabbed some from the spreadsheet, thanks!

**Review by torresga — APPROVED** [review]

**Pull request merged**